tests/unit_test/config.py

```diff
@@ -0,0 +1,296 @@
+import json
+import os
+import subprocess
+
+# Run `terraform plan -out=tfplan` command to generate the plan file
+os.chdir("../../infrastructure/")
+subprocess.run(["terraform", "init"])
+subprocess.run(["terraform", "validate"])
+subprocess.run(
+    ["terraform", "plan", "-var-file=terraform.testsaad.tfvars", "-out=tfplan"]
+)
+
+# Run `terraform show -json tfplan` command to get the plan output in JSON format
+terraform_show_output = subprocess.run(
+    ["terraform", "show", "-json", "tfplan"], capture_output=True, text=True
+).stdout
+
+# Parse the JSON output
+terraform_plan_json = json.loads(terraform_show_output)
+
+most_expensive_service = "most_expensive_service"
+list_linked_accounts = "list_linked_accounts"
+total_account_cost = "total_account_cost"
+project_spend_cost = "project_spend_cost"
+resource_list = "resource_list"
+
+combined_list = []
+for item in terraform_plan_json["planned_values"]["root_module"]["child_modules"]:
+    combined_list.extend(item["resources"])
+
+region = None
+for data_item in combined_list:
+    if data_item["type"] == "aws_cloudtrail" and data_item["name"] == "this":
+        region = data_item["values"]["kms_key_id"].split(":")[3]
+        break
+
+kms_id = None
+for data_item in combined_list:
+    if data_item["type"] == "aws_cloudtrail" and data_item["name"] == "this":
```

Comment:
now we have made a recent change in kms resource please make sure it will not be impacted by that change